src/functions/public/Auth/Revoke-GitHubToken.ps1

```diff
@@ -0,0 +1,136 @@
+
+        .LINK
+        https://psmodule.io/GitHub/Functions/Auth/Revoke-GitHubToken
+    #>
+    [OutputType([void])]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidLongLines', '', Justification = 'Contains a long link.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName = 'OAuth'
+    )]
+    param(
+        # The Client ID of the OAuth application for OAuth token revocation.
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'OAuth'
+        )]
+        [string] $ClientID,
+
+        # The access token to revoke. If not specified, uses the token from the current context.
+        [Parameter(ParameterSetName = 'OAuth')]
+        [Parameter(ParameterSetName = 'Installation')]
+        [string] $Token,
+
+        # Revoke an installation access token instead of an OAuth token.
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'Installation'
+        )]
+        [switch] $InstallationToken,
+
+        # An array of credentials to revoke using the bulk revocation endpoint.
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'Credentials'
+        )]
+        [string[]] $Credentials,
+
+        # The context to run the command in. Used to get the details for the API call.
+        # Can be either a string or a GitHubContext object.
+        [Parameter()]
+        [object] $Context
+    )
+
+    begin {
+        $stackPath = Get-PSCallStackPath
+        Write-Debug "[$stackPath] - Start"
+        $Context = Resolve-GitHubContext -Context $Context
+        Assert-GitHubContext -Context $Context -AuthType IAT, PAT, UAT
+    }
+
+    process {
+        if (-not $Token -and $PSCmdlet.ParameterSetName -ne 'Credentials') {
+            $Token = Get-GitHubAccessToken -Context $Context -AsPlainText
+        }
+
+        if ($PSCmdlet.ParameterSetName -eq 'OAuth') {
+            $inputObject = @{
+                Method      = 'DELETE'
+                APIEndpoint = "/applications/$ClientID/token"
+                Body        = @{
+                    access_token = $Token
+                }
+                Context     = $Context
+            }
+            $targetDescription = "OAuth token for application [$ClientID]"
+        } elseif ($InstallationToken) {
+            $inputObject = @{
+                Method      = 'DELETE'
+                APIEndpoint = '/installation/token'
+                Context     = $Context
+            }
+            $targetDescription = 'Installation access token'
+        } elseif ($PSCmdlet.ParameterSetName -eq 'Credentials') {
+            $inputObject = @{
+                Method      = 'POST'
+                APIEndpoint = '/credentials/revoke'
+                Body        = @{
+                    credentials = $Credentials
+                }
+                Context     = $Context
+            }
+            $targetDescription = "$($Credentials.Count) credential(s)"
+        }
```

Comment:
Split these out to separate private functions (src/functions/private/Auth). Call them something starting with "Remove-GitHub...". When that is done, add them here as references.